src/wasm/wasm-binary.cpp

```diff
@@ -335,10 +335,18 @@ void WasmBinaryWriter::writeFunctions() {
     // Emit Stack IR if present, and if we can
     if (func->stackIR && !sourceMap && !DWARF) {
       BYN_TRACE("write Stack IR\n");
-      StackIRToBinaryWriter(*this, o, func).write();
+      StackIRToBinaryWriter writer(*this, o, func);
+      writer.write();
+      if (debugInfo) {
+        funcMappedLocals[func->name].swap(writer.getMappedLocals());
```

Comment:
Using `std::move` might be clearer here; `swap` makes it look like there might be something useful in the `funcMappedLocals` already. Same below.